### PR TITLE
`assert_equal` is imported top level

### PR DESCRIPTION
**Pull request by astralcai**
**Context**
`assert_equal` basically asserts `qml.equal` but tells you why something isn't equal when it fails.

**Changes**
- Make `qml.ops.functions.equal.assert_equal` accessible as `qml.assert_equal`
- Replaces use of `assert qml.equal()` with `qml.assert_equal()`

[sc-66069]

### CODE REVIEW

File: pennylane/ops/functions/equal.py [hunk 1]
```diff
@@ -202,17 +202,18 @@ def assert_equal(
...
 
         This function is still under developement.
 
-    .. see-also::
+    .. seealso::
 
         :func:`~.equal`
 
+    **Example**
+
```
[the remark this answers is not in the document]
Comment (astralcai) [reply pennylane/ops/functions/equal.py:216]: Done

File: tests/ops/op_math/test_exp.py [hunk 2]
```diff
@@ -546,7 +546,7 @@ def test_trotter_is_used_if_num_steps_is_defined(self):
         ]
         assert len(dec) == len(expected_decomp)
         for op1, op2 in zip(dec, expected_decomp):
-            qml.equal(op1, op2)
+            qml.assert_equal(op1, op2)
```
Comment (albi3ro) [inline tests/ops/op_math/test_exp.py:549]: 🤦

Comment (albi3ro) [reply tests/ops/op_math/test_exp.py:549]: such a useful test.

Comment (astralcai) [reply tests/ops/op_math/test_exp.py:553]: I opened a separate PR to fix this test: https://github.com/PennyLaneAI/pennylane/pull/5867

File: tests/ops/functions/test_commutator.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def test_Hamiltonian_sum(self):
         )
         res = qml.commutator(H1, H2).simplify()
         assert isinstance(res, Sum)
-        assert qml.equal(
+        qml.assert_equal(
             true_res, res
         )  # issue https://github.com/PennyLaneAI/pennylane/issues/5060 as well as potential fix https://github.com/PennyLaneAI/pennylane/pull/5037
```
[the remark this answers is not in the document]
Comment (astralcai) [reply tests/ops/functions/test_commutator.py:73]: Did you actually go through every single file I did a find-replace on? 😮

**Review by albi3ro — APPROVED** [review]
👍

File: tests/ops/functions/test_commutator.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/ops/functions/test_commutator.py]: ...yeah :sweat_smile:

[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/ops/functions/test_commutator.py]: I mean it is visually repetitive, so it's quick to scroll through :)

**Pull request merged**